hypervisor_bridge.c

```diff
@@ -38,6 +39,22 @@
 #include "pcap_capture.h"
 #include "pcap_filter.h"
 
+static int add_nio_desc(nio_t *nio, const char *fmt, ...)
+{
+	int len;
+	va_list argptr;
+
+	va_start(argptr, fmt);
+	len = vsnprintf(NULL, 0, fmt, argptr);
+
+    if (!(nio->desc = malloc((len + 1) * sizeof(char))))
```

Comment:
Ok, thanks for the explanation. After reading some documentation about the memory allocation in C, i've understood the problem: i think we have to free memory right before freeing the nio struct, i'm right?